### PR TITLE
Fixed issue #3087, sneak glitch.

### PR DESCRIPTION
**Pull request by matkal-2**
Me and a couple of student had for assignment to make a change in a public git repository. This is what we came up with.
Comparing to Minecraft you cannot latch on to a block by sneaking if you are already in the air. With this implementation the same goes for Minetest.

Please come with feed-back if you have any.

### CODE REVIEW

**Comment by PilzAdam** [comment]
Only do this if `physics_override_sneak_glitch` is false; this way games and mods can control whether players are allowed to use the glitch.

Also it seems kinda hacky to override the controls; I'm not sure if this is a proper fix.

File: src/localplayer.cpp [hunk 1]
```diff
@@ -176,6 +176,12 @@ void LocalPlayer::move(f32 dtime, Environment *env, f32 pos_max_d,
 	// Maximum distance over border for sneaking
 	f32 sneak_max = BS*0.4;
 
+	// If not touching_ground then we cannot let player sneak
+	if (!touching_ground)
```
Comment (rubenwardy) [inline src/localplayer.cpp]: we do

    if (!touching_ground) {

not 

     if (!touching_ground)
     {

But I guess that can be fixed by someone else, the person that merges.

**Comment by matkal-2** [comment]
I use to put { on the same line as function declaration aswell but I looked through , http://dev.minetest.net/Code_style_guidelines, and it appears that it is not wanted. :)
Mabye wiki is out of date.

**Comment by rubenwardy** [comment]
Functions are the exception, everything else has the brace on the same line.

**Comment by matkal-2** [comment]
I see :)
My bad.

**Comment by est31** [comment]
Agree with both of @PilzAdam's objections.

**Comment by est31** [comment]
I'll close this now, @matkal-2 or anybody else if you want to improve the PR to meet the objections, you can ask for this PR to be reopened, or open a new PR.

**Pull request closed without merging**